### PR TITLE
cleanup inputs and use lib

### PR DESCRIPTION
**Pull request by GTrunSec**
the lib makes it simple and easy.

### CODE REVIEW

**Comment by GTrunSec** [comment]
@danielphan2003  please review it. thx

**Comment by GTrunSec** [comment]
@blaggacao  how about rename the `vs-ext` to `devos-ext-lib` ? we can add the different attrs of overlay extending the pkgs.lib
something like:
```nix
devos-ext-lib.overlay.vscode
devos-ext-lib.overlay.vim
```

File: vscode/overlay.nix [hunk 1]
```diff
@@ -0,0 +1,8 @@
+final: prev: {
+  vscode-utils = prev.vscode-utils // (prev.callPackage ./. { });
+  lib = prev.lib.extend (lfinal: lprev: { } // import ./. {
+    lib = final.lib;
+    vscode-utils = final.vscode-utils;
+    vscode = final.vscode;
```
Comment (danielphan2003) [inline vscode/overlay.nix]:
```suggestion
    inherit (final) lib vscode-utils vscode;
```

**Comment by danielphan2003** [comment]
Thanks for the changes, this has been way over due on my side.

**Comment by danielphan2003** [comment]
> @blaggacao how about rename the `vs-ext` to `devos-ext-lib` ? we can add the different attrs of overlay extending the pkgs.lib something like:
> 
> ```nix
> devos-ext-lib.overlay.vscode
> devos-ext-lib.overlay.vim
> ```


`devos-ext-lib` sounds good to me. Given the simplicity of this repo, `vim-ext` should be right at home with `vs-ext`.

**Comment by blaggacao** [comment]
Merge this first? Then rename?

**Comment by GTrunSec** [comment]
> Merge this first? Then rename?

yeh，can you do that?

File: vscode/overlay.nix (hunk 1 above)
Comment (GTrunSec) [reply vscode/overlay.nix]: thx

**Comment by blaggacao** [comment]
Renamed to `devos-ext-lib` :rocket:

**Pull request merged**